### PR TITLE
feat(link-with-icon): added icon-placement property

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
Refs #4114.

### Description

Added an attribute that defines the side where the icon should be rendered

### CODE REVIEW

File: packages/web-components/src/components/link-with-icon/link-with-icon.ts [hunk 1]
```diff
@@ -26,13 +34,19 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-link-with-icon`)
 class DDSLinkWithIcon extends StableSelectorMixin(DDSLink) {
+  /**
+   * Defines the position of the icon
+   * `left` or `right`
+   */
+  @property({ reflect: true, attribute: 'icon-placement' })
```
Comment (asudoh) [inline packages/web-components/src/components/link-with-icon/link-with-icon.ts:41]: For aligning dash-cased attribute name and camel-cased property name. An alternate approach is renaming `linkIconPlacement` with `iconPlacement`:

```suggestion
  @property({ reflect: true, attribute: 'link-icon-placement' })
```

**Review by asudoh — COMMENTED** [review]
Great change, just one comment - Thank you for doing this @RaphaelAmadeu!

File: packages/web-components/src/components/link-with-icon/link-with-icon.ts (hunk 1 above)
Comment (RaphaelAmadeu) [reply packages/web-components/src/components/link-with-icon/link-with-icon.ts]: @asudoh this was addressed in my latest commit

File: packages/web-components/src/components/link-with-icon/link-with-icon.ts [hunk 2]
```diff
@@ -26,13 +34,19 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-link-with-icon`)
 class DDSLinkWithIcon extends StableSelectorMixin(DDSLink) {
+  /**
+   * Defines the position of the icon
+   * `left` or `right`
+   */
+  @property({ reflect: true, attribute: 'link-icon-placement' })
+  linkIconPlacement = LINK_ICON_PLACEMENT_TYPES.DEFAULT;
```
Comment (asudoh) [inline packages/web-components/src/components/link-with-icon/link-with-icon.ts:42]:
```suggestion
  linkIconPlacement = LINK_ICON_PLACEMENT_TYPES.RIGHT;
```

File: packages/web-components/src/components/link-with-icon/link-with-icon.ts [hunk 3]
```diff
@@ -17,6 +17,14 @@ import styles from './link-with-icon.scss';
 const { prefix } = settings;
 const { stablePrefix: ddsPrefix } = ddsSettings;
 
+/**
+ * Enum for the icon placement possibilities
+ */
+export enum LINK_ICON_PLACEMENT_TYPES {
+  DEFAULT = 'left',
```
Comment (asudoh) [inline packages/web-components/src/components/link-with-icon/link-with-icon.ts:24]:
```suggestion
  LEFT = 'left',
```

File: packages/web-components/src/components/link-with-icon/__stories__/link-with-icon.stories.ts [hunk 4]
```diff
@@ -35,6 +45,12 @@ export default {
         disabled: boolean('Disabled (disabled)', false, groupId),
         href: textNullable('Link href (href)', 'https://github.com/carbon-design-system/carbon-web-components', groupId),
         onClick: action('click'),
+        iconPlacement: select(
+          'Icon Position (icon-placement):',
+          placementTypes,
+          placementTypes[`${LINK_ICON_PLACEMENT_TYPES.DEFAULT}`],
```
Comment (asudoh) [inline packages/web-components/src/components/link-with-icon/__stories__/link-with-icon.stories.ts:51]:
```suggestion
          placementTypes[`${LINK_ICON_PLACEMENT_TYPES.RIGHT}`],
```

File: packages/web-components/src/components/link-with-icon/__stories__/link-with-icon.stories.ts [hunk 5]
```diff
@@ -9,22 +9,32 @@
...
+      href="${ifNonNull(href)}"
+      @click="${onClick}"
+    >
       ${children}${ArrowRight20({ slot: 'icon' })}
     </dds-link-with-icon>
   `;
 };
 
+const placementTypes = {
+  [`${LINK_ICON_PLACEMENT_TYPES.DEFAULT}`]: LINK_ICON_PLACEMENT_TYPES.DEFAULT,
```
Comment (asudoh) [inline packages/web-components/src/components/link-with-icon/__stories__/link-with-icon.stories.ts:34]:
```suggestion
  [`${LINK_ICON_PLACEMENT_TYPES.LEFT}`]: LINK_ICON_PLACEMENT_TYPES.LEFT,
```

File: packages/web-components/src/components/link-with-icon/link-with-icon.ts [hunk 6]
```diff
@@ -43,14 +57,18 @@ class DDSLinkWithIcon extends StableSelectorMixin(DDSLink) {
...
-      <slot name="icon"></slot>
+      <slot class="${prefix}--link-with-icon__icon-${this.linkIconPlacement}" name="icon"></slot>
     `;
   }
 
   protected _renderInner() {
-    return html`
-      ${this._renderContent()}${this._renderIcon()}
-    `;
+    return this.linkIconPlacement === LINK_ICON_PLACEMENT_TYPES.DEFAULT
```
Comment (asudoh) [inline packages/web-components/src/components/link-with-icon/link-with-icon.ts:65]:
```suggestion
    return this.linkIconPlacement === LINK_ICON_PLACEMENT_TYPES.LEFT
```

**Review by asudoh — COMMENTED** [review]
Thank you for the update @RaphaelAmadeu! I suspect making the right placement as the default eliminates a bunch of Percy diffs, and probably also that was why we got a bunch of snapshot changes:

**Review by asudoh — APPROVED** [review]
LGTM 👍, just one question - Is the Percy diff intentional? Thanks @RaphaelAmadeu!

**Comment by jeffchew** [comment]
> LGTM 👍, just one question - Is the Percy diff intentional? Thanks @RaphaelAmadeu!

I don't see an icon placement knob for link list stories for web components, but we have that for React. What is the default placement @RaphaelAmadeu ?

**Pull request merged**